### PR TITLE
Fix shared library build and introduce FAST

### PR DESCRIPTION
**Pull request by theopolis**
For faster iterative builds one can set `FAST=1` while making.

### CODE REVIEW

**Comment by theopolis** [comment]
I enabled OSX 10.9 within this PR mistakenly. I have a staged branch that targets fixing 10.9, but this will receive a FN.

**Comment by osqueryer** [comment]
:-1: The commit 169364866dfd4d6592393f111aac005ef34e38e0 (Job results: [3008](https://jenkins.osquery.io/job/osqueryPullRequestBuild/3008/)) failed one or more tests.

File: osquery/CMakeLists.txt [hunk 1]
```diff
@@ -253,6 +260,9 @@ if(NOT ${OSQUERY_BUILD_SDK_ONLY})
   if(DEFINED ENV{OSQUERY_BUILD_SHARED})
     target_link_libraries(daemon libosquery_shared)
     target_link_libraries(daemon libosquery_additional_shared)
+    if(DEFINED ENV{FAST})
+      target_link_libraries(shell "-Wl,-rpath,${CMAKE_BINARY_DIR}/osquery")
```
Comment (theopolis) [inline osquery/CMakeLists.txt]: This should be `daemon`.

**Comment by osqueryer** [comment]
:+1: The commit 970870e0cee9f97eed9eb37ea4efb7df69410894 (Job results: [3022](https://jenkins.osquery.io/job/osqueryPullRequestBuild/3022/)) was built successfully!

**Pull request merged**